src/AppBundle/ControllerTemplates/JSONController.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/*
+ * Intend to use this to store boilerplate HTTP things to do with manipulating JSON data.
+ */
+
+namespace AppBundle\ControllerTemplates;
+
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * A exception thrown by JSONController.
+ *
+ * Codes:
+ * 0: generic
+ * 1: Parser error
+ * 2: Unexpected exception
```

Comment:
You should really have three different exceptions to account for this, instead of having to pass in a magic number that tells you what caused the exception.